plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/impl/api/groupBy.kt

```diff
@@ -321,129 +313,96 @@ class GroupBySumOf : GroupByAggregatorSumOf(defaultName = "sum")
  * - [name] Optional name for the resulting aggregated column. Defaults to `defaultName`.
  * - [columns] ColumnsResolver to define which columns to include in the grouping operation.
  */
-abstract class GroupByAggregator0(val defaultName: String) : AbstractSchemaModificationInterpreter() {
+abstract class GroupByAggregator0(val defaultName: String, val aggregator: Aggregator<*, *>) : AbstractSchemaModificationInterpreter() {
     val Arguments.receiver by groupBy()
     val Arguments.name: String? by arg(defaultValue = Present(null))
     val Arguments.columns: ColumnsResolver? by arg()
 
     override fun Arguments.interpret(): PluginDataFrameSchema {
         if (name == null) {
-            val resolvedColumns = columns?.resolve(receiver.keys)?.map { it.column }!!.toList()
-            return PluginDataFrameSchema(receiver.keys.columns() + resolvedColumns)
+            val resolvedColumns = columns?.resolve(receiver.keys)?.map { (it.column as SimpleDataColumn) }!!.toList()
```

Comment:
why columns is nullable? i think you can `ColumnsResolver?` => `ColumnsResolver` and remove !!